bots/tests-policy

```diff
@@ -111,6 +121,55 @@ def filterRetry(output, prefix):
             lines[i] = prefix
     return "\n".join(lines)
 
+# Figure out the name from a failed test
+def parseName(output):
+    for line in output.split("\n"):
+        if line.startswith("not ok "):
+            line = line[7:]
+            while line[0].isspace() or line[0].isdigit():
+                line = line[1:]
+            (name, delim, directive) = line.partition("#")
+            (name, delim, directive) = name.partition("duration")
+            name = name.strip()
+            return name
+    return ""
+
+# -----------------------------------------------------------------------------
+# Flakiness Checks
+
+def guessFlake(output, context):
+    if not WITH_LEARNING:
+        return output
+
+    path = os.path.join(DATA, task.learn1.LEARN_DATA)
+    if not os.path.exists(path):
+        return output
+
+    # The pickled neural network
+    network = task.learn1.load(path)
+
+    # Build up an item just like in tests-data
+    item = {
+        "pull": None,
+        "revision": os.environ.get("TEST_REVISION"),
+        "status": "failure",
+        "context": context,
+        "date": time.strftime("%Y-%m-%dT%H:%M:%SZ"),
+        "merged": None,
+        "test": parseName(output),
+        "url": None,
+        "log": output
+    }
+
+    pred_proba = network.predict_proba(item)
+    if max(pred_proba) >= 0.5:
+        if int(pred_proba[1] >= pred_proba[0]) == 1:
```

Comment:
The `int(…) == 1` should be replaced with `…`, but with a construct like that it's actually likely that the author meant something entirely different.